src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1292,5 +1337,30 @@ public Worker build() {
                     workerStateChangeListener);
         }
 
+        AmazonWebServiceClient createClient(@NonNull final AwsClientBuilder builder,
+                                            final AWSCredentialsProvider credentialsProvider,
+                                            final ClientConfiguration clientConfiguration,
+                                            final String endpointUrl,
+                                            final String region) {
+            if (credentialsProvider != null) {
+                builder.withCredentials(credentialsProvider);
+            }
+            if (clientConfiguration != null) {
+                builder.withClientConfiguration(clientConfiguration);
+            }
+            if (StringUtils.isNotEmpty(endpointUrl)) {
+                LOG.warn("Received configuration for endpoint as " + endpointUrl + ", and region as "
+                        + region + ".");
+                builder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpointUrl, region));
+            } else if (StringUtils.isNotEmpty(region)) {
+                LOG.warn("Received configuration for region as " + region + ".");
+                builder.withRegion(region);
+            } else {
+                LOG.warn("No configuration received for endpoint and region, will default region to us-east-1");
+                builder.withRegion(Regions.US_EAST_1);
+            }
+            return (AmazonWebServiceClient) builder.build();
```

Comment:
You might be able to get away without the cast if change the signature to something like:
``` java
<R, T extends AwsClientBuilder<T, R>> R setupClient(T builder, AWSCredentialsProvider creds ...) {
    return builder.withCredentials(creds).build();
}
```